src/hoc/withTheme.js

```diff
@@ -23,25 +22,46 @@ const wrapWithTheme = (Component: ReactClass<any>) => {
     static styledComponentId = 'withTheme'
 
     static contextTypes = {
-      [CHANNEL]: PropTypes.func,
+      [CHANNEL]: CONTEXT_CHANNEL_SHAPE,
```

Comment:
No because people using that version of styled-components won't be able to theme component libraries built with an earlier version so this is a breaking change.